### PR TITLE
feat: [CDS-56775]: added selective stage exec on triggers

### PR DESCRIPTION
**Pull request by sahithikolichala**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by sahithikolichala** [comment]
retrigger jest

File: src/modules/72-triggers/pages/triggers/views/WebhookPipelineInputPanel.tsx [hunk 1]
```diff
@@ -189,7 +189,10 @@ function WebhookPipelineInputPanelForm({
       parentEntityRepoName: repoName
     },
     body: {
-      stageIdentifiers: []
+      stageIdentifiers:
+        formikProps.values?.stagesToExecute && formikProps.values?.stagesToExecute.length
+          ? [...formikProps.values.stagesToExecute]
```
Comment (rameshx) [inline src/modules/72-triggers/pages/triggers/views/WebhookPipelineInputPanel.tsx:194]: Nit: no need to spread to a new array

File: src/modules/72-triggers/pages/triggers/__tests__/TriggersWizardPage.test.tsx [hunk 2]
```diff
@@ -99,6 +99,12 @@ jest.mock('services/cd-ng-rq', () => ({
   })
 }))
 
+// jest.mock('services/pipeline-ng', () => ({
+//   useGetStagesExecutionList: jest.fn().mockImplementation(() => {
+//     return { data: [], refetch: jest.fn() }
+//   })
+// }))
+
```
Comment (rameshx) [inline src/modules/72-triggers/pages/triggers/__tests__/TriggersWizardPage.test.tsx:107]: remove if not needed

File: src/modules/72-triggers/components/steps/ArtifactTriggerInputPanel/ArtifactTriggerInputPanel.tsx [hunk 3]
```diff
@@ -99,7 +99,10 @@ function ArtifactTriggerInputPanelForm({
       branch
     },
     body: {
-      stageIdentifiers: []
+      stageIdentifiers:
+        formikProps.values?.stagesToExecute && formikProps.values?.stagesToExecute.length
+          ? [...formikProps.values.stagesToExecute]
+          : []
```
Comment (rameshx) [inline src/modules/72-triggers/components/steps/ArtifactTriggerInputPanel/ArtifactTriggerInputPanel.tsx:105]:
```suggestion
      stageIdentifiers: formikProps?.values?.stagesToExecute ?? []
```

**Comment by sahithikolichala** [comment]
retrigger cypress-cd

**Review by pankajsingh-harness — APPROVED** [review]

**Comment by sahithikolichala** [comment]
retrigger jest

**Comment by sahithikolichala** [comment]
retrigger cypress-cd

**Comment by sahithikolichala** [comment]
retrigger titlecheck

**Comment by sahithikolichala** [comment]
unrelated tests are failing
Screen Shot 2023-04-19 at 10 10 31 PM

**Pull request merged**